webapp/app.py

```diff
@@ -113,28 +130,39 @@ def logout():
     """
     Empty the session, used to logout.
     """
-    flask.session.pop("openid", None)
+    flask.session.pop("authentication_token", None)
+    flask.session.pop("macaroon_root", None)
+    flask.session.pop("macaroon_discharge", None)
 
     return flask.redirect(open_id.get_next_url())
 
 
 @app.route("/login", methods=["GET", "POST"])
 @open_id.loginhandler
 def login_handler():
-    if "openid" in flask.session and "macaroon_root" in flask.session:
+    if (
+        "authentication_token" in flask.session
+        and "macaroon_root" in flask.session
```

Comment:
We should only check for `authentication_token` right? I don't think the other two will exist will they?